nginx.config/default.conf

```diff
@@ -0,0 +1,10 @@
+server {
+    listen 80;
+
+    location / {
+        proxy_buffering off;
+        root /usr/share/nginx/html;
+        try_files $uri $uri/ $uri/index.html /index.html =404;
+    }
+}
+
```

Comment:
Default nginx `default.conf` file looks like that:
```
server {
    listen       80;
    server_name  localhost;

    #access_log  /var/log/nginx/host.access.log  main;

    location / {
        root   /usr/share/nginx/html;
        index  index.html index.htm;
    }
    error_page   500 502 503 504  /50x.html;
    location = /50x.html {
        root   /usr/share/nginx/html;
    }

}
```
What is different and really required in your version?

I suppose when there is no `proxy_pass` or similar config proxy is not active? Does this option affect non-proxy requests?

`root` is the same config.
`try_files` is the only difference. I think the use-case is when you are in a sub-page (e.g. `/edit/connector` or something) and the URI in the browser isn't `/` anymore and you then reload the page?
@indraraj is it okay to have a full URI in the browser and should traffic be re-directed to `index.html` then or would we move the URI behind an anchor (e.g. `#/edit/connector`)? Sorry, currently I am not aware of the URIs that we use in the UI frontend.
5xx error config is missing in this version.

The only difference I could spot is try_files and I see a potential use-case for it.
Instead of changing the original `default.conf` maybe we can just add another `conf.d/debezium-ui.conf` to add missing options?

Could you also file a DBZ task under https://issues.redhat.com/ describing the issue and prefix the commit with this?
@ani-sha @indraraj we need to recover the previous checks on PRs. I think it got lost with the removal of the UI backend.

Reply:
When a user navigates to a specific route in your React application, the browser sends a request to the server for that route. Since Nginx doesn't have a corresponding file or directory for that route (because it's a single-page application), it would normally return a 404 error.

However, by using the `try_files` directive, you can instruct Nginx to always serve the main HTML file (which typically contains the JavaScript bundle for your React application) for any requested URL. This allows your React application to handle client-side routing internally, based on the URL, without needing to make additional requests to the server for each route. For importing the old config and add them for the new one there is no reason behind it but both options we can consider it as okay.

Reply:
Yep, gotcha! That's how I understood your change. So `try_files` is the only addition, right? I wonder what is the correct way to customize (the default) config of nginx for this. I think the best way is adding a new file like I suggested above `conf.d/debezium-ui.conf`. But I need to figure out what needs to be in there. Would be great if you could figure this out @yahyaAlsaidi? Or what do you think?

Reply:
@rk3rn3r from server side we just load the single js bundle on the main domain and the subsequent route are handled on the client side in the SPA react application.

we have to perform some work in modifying the remaining previous checks(like E2E) that we had earlier and for them to work after update. I will log a jira issue(if its not already there) then we can check in the team call if or when we need to pick them.

Reply:
Creating new files is beneficial when managing complex setups with multiple server blocks and numerous location directives. In such cases, organizing configurations into separate files and then including them in the default file can enhance clarity and ease of management. However, in scenarios where there is only one server block and a limited number of location directives to handle, adding a new file might unnecessarily complicate the configuration process. It could lead to confusion and make it more challenging to understand and follow the configuration. In simpler setups like this, directly modifying the existing configuration file is often more straightforward and easier to comprehend. @rk3rn3r If you wish me to do it for you I can do it with only adding the include command in the nginx file here is way of doing it https://serverfault.com/questions/707955/nginx-split-large-configuration-file you can check this one if you want me to do it ?

Reply:
Thx @yahyaAlsaidi for the exploration. I understand and agree to that statement when looking at a specific server's config. My question was a bit more towards dependency management from the UI Nginx container to the upstream Nginx base container. I don't want to be forced to update the default.conf when it was updated upstream. Instead I thought we just keep the original file inherited from the base repo and we only add our additional bits in a new file.
That way the base config is well separated from the UI specific one.
I hope you understand my concern?

Reply:
@indraraj but what happens on reload after you navigated to a sub-page/URL? The frontend app that can only be loaded via `index.html` in the current nginx configuration will not be loaded from a sub-URL, right? Then we need this change that @yahyaAlsaidi suggest with this PR in order to correctly load the UI.
BTW, @yahyaAlsaidi @indraraj, the updated nginx config will load the UI and render the correct sub-page that is referenced by the URL?

Reply:
Yes, this fix we let you reload the page with the sub-page that is referenced by the URL.

Reply:
yes, from the server we need to load the same JS build for all the paths, it's in UI that then handles the exact URL present in the browser to load the necessary UI page for it. I have tested it locally this is fixing the reload issue on suburls.